HedgeModManager/UI/MainWindow.xaml.cs

```diff
@@ -1431,22 +1431,55 @@ private void UI_Download_Codes(object sender, RoutedEventArgs e)
             UpdateStatus(string.Format(Localise("StatusUIDownloadingCodes"), HedgeApp.CurrentGame));
             try
             {
-                var downloader = new DownloadWindow($"Downloading codes for {HedgeApp.CurrentGame}", HedgeApp.CurrentGame.CodesURL, CodeProvider.CodesTextPath)
+                string codesPath = CodeProvider.CodesTextPath;
+
+                /* Parse current codes list, since ModsDB.CodesDatabase
+                   is contaminated with codes from ExtraCodes.hmm */
+                var oldCodes = new CodeFile(codesPath);

```

Comment:
It might be worth checking if the file at codesPath exists here, we don't want to show a massive diff to the user if they didn't have codes.